frescobaldi_app/quickinsert/spanners.py

```diff
@@ -283,10 +278,16 @@ def spanner_positions(cursor):
         partial = ly.document.OUTSIDE
 
     items = list(ly.rhythm.music_items(c, partial=partial))
-    if cursor.hasSelection():
-        del items[1:-1]
+
+    if command.endswith('melisma'):
```

Comment:
thought: At first glance it seems better to add an own function for melismas than to add a special case in this function.

Reply:
I thought so too, but being the first time I intervened on the code I didn't dare to extrapolate the command into a separate function. An independent function would certainly make everything clearer.
I could improve it by putting melisma insertion in its own function.